src/plugins/expressions/common/expression_renderers/types.ts

```diff
@@ -63,20 +64,20 @@ export interface IInterpreterRenderHandlers {
   /**
    * Done increments the number of rendering successes
    */
-  done: () => void;
-  onDestroy: (fn: () => void) => void;
-  reload: () => void;
-  update: (params: any) => void;
-  event: (event: any) => void;
-  hasCompatibleActions?: (event: any) => Promise<boolean>;
-  getRenderMode: () => RenderMode;
+  done(): void;
```

Comment:
I think it is not. Otherwise, that should be fixed there. Starting from TS 2.6, it has [stricter](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-6.html#strict-function-types) type checks for the functions. When we define those methods as properties, which is unsound already, we are not able to extend or implement that interface with a subtype.

The following example will throw a compilation error:
```typescript
interface A {
  foo: (bar: unknown) => number;
}

const a: A = {
  foo: (bar: number) => {
    return bar;
  },
};

class B implements A {
  foo(bar: number) {
    return bar;
  }
}
```

But this one will not:
```typescript
interface A {
  foo(bar: unknown): number;
}

const a: A = {
  foo: (bar: number) => {
    return bar;
  },
};

class B implements A {
  foo(bar: number) {
    return bar;
  }
}
```

I have fixed that for the methods with any, but since there are more methods in the interface, I've corrected them all for the consistency.